src/main/java/uk/gov/hmcts/darts/dailylist/controller/DailyListController.java

```diff
@@ -95,12 +97,12 @@ public ResponseEntity<Void> dailylistsHousekeepingPost() {
 
     @Override
     public ResponseEntity<Void> dailylistsRunPost(String listingCourthouse) {
-        if (listingCourthouse == null) {
-            CompletableFuture.runAsync(processor::processAllDailyLists);
-        } else {
-            CompletableFuture.runAsync(() -> processor.processAllDailyListForListingCourthouse(listingCourthouse));
+        var taskName = PROCESS_DAILY_LIST_TASK_NAME.getTaskName();
+        Optional<AutomatedTaskEntity> automatedTaskEntity = automatedTasksApi.getTaskByName(taskName);
+        if (automatedTaskEntity.isPresent() && automatedTasksApi.isLocked(automatedTaskEntity.get())) {
+            throw new DartsApiException(DAILY_LIST_ALREADY_PROCESSING);
         }
-
+        CompletableFuture.runAsync(() -> processor.processAllDailyListsWithLock(listingCourthouse));
```

Comment:
Is there an opportunity for a race condition if the automated task takes the lock between `automatedTasksApi.isLocked()` and `processor.processAllDailyListsWithLock()`? Unlikely I suppose, but what would happen?

Reply:
Yeah I guess there is a slight chance of it happening. To be honest the chances are so slim and if it did happen then no harm would be done, just the Daily list running twice.